src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -326,6 +326,11 @@ The .NET Foundation licenses this file to you under the MIT license.
 
   </Target>
 
+  <PropertyGroup Condition="'$(NativeCodeGen)' == 'llvm'">
+    <EmccOptimizationSetting Condition="$(Optimize) == 'true' and $(IlcOptimizationPreference) != 'Size'">-O3</EmccOptimizationSetting>
```

Comment:
IlcOptimizationPreference was renamed to OptimizationPreference upstream. Something to keep in mind for next integration.